nixos/modules/hardware/printers.nix

```diff
@@ -0,0 +1,111 @@
+{ config, lib, pkgs, services, ... }:
+with lib;
+let
+  cfg = config.hardware.printers;
+  ensurePrinter = p: ''
+    ${pkgs.cups}/bin/lpadmin -p '${p.name}' -E ${optionalString (p.location != null) "-L '${p.location}'"} ${optionalString (p.description != null) "-D '${p.description}'"} -v '${p.deviceUri}' -m '${p.model}'
+  '';
+  ensureDefaultPrinter = name: ''
+    ${pkgs.cups}/bin/lpoptions -d '${name}'
+  '';
+
+  forbiddenChars  = str: filter (c: c == "#" || c == "/") (stringToCharacters str) == [];
+  # TODO: is it possible to combine the forbiddenChars with that regex?
+  # test by using `nix repl` and then `builtins.match "[[:graph:]]+" "teststring#"`
+  printerName = (types.addCheck (types.strMatching "[[:graph:]]+") forbiddenChars)
+    // { description = "printable string without spaces, # and /"; };
+
+
+in
+{
+  options = {
+    hardware.printers = {
+      ensureDefaultPrinter = mkOption {
+        type = types.nullOr printerName;
+        default = null;
+        description = ''
+          Ensures the named printer is the default CUPS printer / printer queue.
+        '';
+      };
+      ensurePrinters = mkOption {
+        description = ''
+          Will regularly ensure that the given CUPS printers are configured as declared here.
+          If a printer's options are manually changed afterwards, they will be overwritten eventually.
+          This option will never delete any printer, even if removed from this list.
+          You can check existing printers with <literal>lpstat -s</literal>
+          and remove printers with <literal>lpadmin -x \<printer-name\></literal>.
```

Comment:
There's `<command>..</command>` for that, also in `deviceUri` and `model`

Reply:
Thank you!